app/addons/documents/resources.js

```diff
@@ -216,11 +214,18 @@ function(app, FauxtonAPI, Documents, PagingCollection) {
       } else if (context === "apiurl"){
         startOfUrl = window.location.origin;
       }
-      var design = app.utils.safeURLName(this.design),
-          view = app.utils.safeURLName(this.view);
 
-      var url = [startOfUrl, this.database.safeID(), "_design", design, this.idxType, view];
-      return url.join("/") + query;
+      var database = this.database.safeID(),
+          design = app.utils.safeURLName(this.design),
+          view = app.utils.safeURLName(this.view),
+          url = FauxtonAPI.urls('view', 'server', database, design, view);
+
+      if (url === '') {
```

Comment:
I'd definitely look into this chunk. Does seem a bit fishy that url *may* be empty. Is that possible, with the new FauxtonAPI.urls method?